libcodechecker/server/api/report_server.py

```diff
@@ -2109,6 +2111,26 @@ def __store_reports(self, session, report_dir, source_root, run_id,
         already_added = set()
         new_bug_hashes = set()
 
+        # Get checker names which was enabled during the analysis.
+        enabled_checkers = set()
+        disabled_checkers = set()
+        for analyzer_checkers in checkers.values():
+            if isinstance(analyzer_checkers, dict):
+                for checker_name, enabled in analyzer_checkers.iteritems():
+                    if enabled:
+                        enabled_checkers.add(checker_name)
+                    else:
+                        disabled_checkers.add(checker_name)
+            else:
+                enabled_checkers.update(analyzer_checkers)
+
+        def checker_is_unavailable(checker_name):
+            """
+            Returns True if the given checker is unavailable.
+            """
```

Comment:
Please add a comment why we filter out `clang-diagnostic-` "checkers". These are warnings and the warning list is not available right now. Using the diagtool could be a solution later so the client can send the warning list to the server.